Webhook unit test panics with invalid memory address

**Describe the bug**
Unit tests for webhook signals panic with the following stacktrace:

```
=== RUN   TestSignal/delete
2018/07/23 04:23:25 server successfully shutdown
2018/07/23 04:23:25 received a request from 'localhost:9000'
2018/07/23 04:23:25 server successfully shutdown
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x831f6e]
goroutine 10 [running]:
testing.tRunner.func1(0xc420240000)
        /usr/local/go/src/testing/testing.go:711 +0x5d9
panic(0x17ad5a0, 0x22762b0)
        /usr/local/go/src/runtime/panic.go:491 +0x2a2
net/http.(*Server).Shutdown(0xc420274000, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/net/http/server.go:2506 +0x20e
github.com/argoproj/argo-events/signals/webhook.(*webhook).Stop(0xc4203faa20, 0xc420124400, 0xc4203ece10)
        /go/src/github.com/argoproj/argo-events/signals/webhook/signal.go:105 +0x89
github.com/argoproj/argo-events/signals/webhook.makeAPIRequest(0xc420240000, 0x18f0592, 0x6, 0x18f16cf, 0x7)
        /go/src/github.com/argoproj/argo-events/signals/webhook/signal_test.go:70 +0x72d
github.com/argoproj/argo-events/signals/webhook.testDeleteRequest(0xc420240000)
        /go/src/github.com/argoproj/argo-events/signals/webhook/signal_test.go:85 +0x63
testing.tRunner(0xc420240000, 0x1970078)
        /usr/local/go/src/testing/testing.go:746 +0x16d
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:789 +0x569
FAIL    github.com/argoproj/argo-events/signals/webhook 0.559s
```

**To Reproduce**
http://argo.applatix.net/workflows/default/argo-events-ci-cprp2?tab=workflow&nodeId=argo-events-ci-cprp2-613907525&sidePanel=logs%3Aargo-events-ci-cprp2-613907525%3Amain

**Expected behavior**
Unit tests should succeed.

--- a/signals/webhook/webhook_test.go
+++ b/signals/webhook/webhook_test.go
@@ -26,39 +26,35 @@ import (
 )
 
 var (
-	testingTargetPort = 45677
-	client            = &http.Client{}
-	payload           = "{name: x}"
+	client  = &http.Client{}
+	payload = "{name: x}"
 )
 
 func handleEvent(t *testing.T, testEventChan <-chan *v1alpha1.Event) {
 	event := <-testEventChan
 
-	if event.Context.Source.Host != fmt.Sprintf("localhost:%d", testingTargetPort) {
-		t.Errorf("event Context SourceHost:\nexpected: %s\nactual: %s", fmt.Sprintf("localhost:%d", testingTargetPort), event.Context.Source.Host)
-	}
 	if string(event.Data) != payload {
 		t.Errorf("event Data:\nexpected: %s\nactual: %s", payload, string(event.Data))
 	}
 }
 
-func makeAPIRequest(t *testing.T, httpMethod string, endpoint string) {
+func makeAPIRequest(t *testing.T, httpMethod string, endpoint string, port int32) {
 	web := New()
 	signal := v1alpha1.Signal{
 		Webhook: &v1alpha1.WebhookSignal{
-			Port:     int32(testingTargetPort),
+			Port:     port,
 			Endpoint: endpoint,
 			Method:   httpMethod,
 		},
 	}
 	done := make(chan struct{})
 	// stop listening and ensure the events channel is closed on exit
-	defer func() { done <- struct{}{} }()
+	defer close(done)
 	events, err := web.Listen(&signal, done)
 
 	go handleEvent(t, events)
 
-	request, err := http.NewRequest(httpMethod, fmt.Sprintf("http://localhost:%d%s", testingTargetPort, endpoint), strings.NewReader(payload))
+	request, err := http.NewRequest(httpMethod, fmt.Sprintf("http://localhost:%d%s", port, endpoint), strings.NewReader(payload))
 	if err != nil {
 		t.Fatalf("unable to create http request. cause: %s", err)
 	}
@@ -73,15 +69,15 @@ func makeAPIRequest(t *testing.T, httpMethod string, endpoint string) {
 }
 
 func testPostRequest(t *testing.T) {
-	makeAPIRequest(t, http.MethodPost, "/post")
+	makeAPIRequest(t, http.MethodPost, "/post", 5677)
 }
 
 func testPutRequest(t *testing.T) {
-	makeAPIRequest(t, http.MethodPut, "/put")
+	makeAPIRequest(t, http.MethodPut, "/put", 5678)
 }
 
 func testDeleteRequest(t *testing.T) {
-	makeAPIRequest(t, http.MethodDelete, "/delete")
+	makeAPIRequest(t, http.MethodDelete, "/delete", 5679)
 }
 
 func TestSignal(t *testing.T) {

